ADR/0008-integration-service-api-contracts.md

```diff
@@ -0,0 +1,256 @@
+# 8. HACBS Test Stream - API contracts
+
+Date: 2022-11-28
+
+## Status
+
+Accepted
+
+## Context
+
+The Hybrid Application Cloud Build Service being developed aims to serve Red Hat teams but also partners and customers. This requires a level of adaptability to avoid recreating custom flows and Tasks for each stakeholder.
+
+In this respect Tasks developed by HACBS test stream should allow swapping external systems to accommodate different environments. This swap should not induce the complete recreation of pipelines.
+
+This and the idea of providing a homogeneous experience, which is easier to comprehend and navigate complex systems, leads to the definition of `API contracts`. These contracts need to be understood as guidance that may evolve with time and experience while keeping the aim of building a flexible homogeneous system.
+
+Tasks from the test stream may exchange information through
+
+1. Parameters passed at the time of the trigger of a PipelineRun (that is user, git repository, etc)
+2. Output of upstream elements of the Pipeline which may be needed for performing the Task (that is build artifacts, container image, etc)
+3. Providing whether the Task was successful or not
+4. Output of the Task that may be needed for processing subsequent Tasks of the pipeline
+5. Output of the Task that need to be stored for auditing and troubleshooting (that is summarized results of a validation or a scan, etc)
+6. Communication with external systems that are needed for performing the Task (that is container image repository, image scanner, etc)
+7. Connection details and credentials that may be required for the above
+
+Related to [PLNSRVCE-41](https://issues.redhat.com/browse/PLNSRVCE-41) investigations in regards to [Tekton results](https://github.com/tektoncd/results). The investigations are documented [here](https://docs.google.com/document/d/1WGWx6MllVs-BwRLd0PP5QTQHgGp0fW2dOYRKRAKWiwo/edit#heading=h.bpyxfhyud4dv).
+
+## Decision
+
+### Results and Records
+
+The output of each Tekton task will be provided in two forms:
+- As a minimized [Tekton result](https://tekton.dev/docs/pipelines/tasks/#emitting-results) in JSON format listing all test failures
+  - The name of the result will be **HACBS_TEST_OUTPUT**
+- As a Test output JSON file with full test information that’s saved in the Tekton Pipeline Workspace
+  - The name of the file will be in snake case and will be in the form of **test_name_output.json**
+
+The maximum size of a [Task's Results](https://tekton.dev/vault/pipelines-v0.17.3/tasks/#emitting-results) is limited by the container [termination message](https://kubernetes.io/docs/tasks/debug/debug-application/determine-reason-pod-failure/#customizing-the-termination-message) feature of Kubernetes.
```

Comment:
No, user can not input message for `note` now. The `skip` often means the test is not run because there is not supported files, such as https://github.com/redhat-appstudio/build-definitions/blob/main/task/sast-snyk-check/0.1/sast-snyk-check.yaml#L48 and https://github.com/redhat-appstudio/build-definitions/blob/main/task/sast-java-sec-check/0.1/sast-java-sec-check.yaml#L66

If the result is longer than 4096 Bytes, the task will meet error `Termination message is above max allowed size 4096, caused by large task result`.

Is a longer namespace meaningful for the team(EC?) consuming these fields? We may confirm with them.

Reply:
I assume that a long namespace field would just be there if the test suite had a long name. I don't think there is any inherent (dis)advantage to a long or short name.

Reply:
Thanks, the normal namespace(The package name of Policy https://github.com/redhat-appstudio/hacbs-test/blob/main/policies/clair/vulnerabilities-check.rego#L1) is supported and will not exceed the limit.